src/url-replacements.js

```diff
@@ -239,6 +242,14 @@ class UrlReplacements {
       return this.getTimingData_('navigationStart',
           'domContentLoadedEventStart');
     });
+
+    // Returns an identifier for the viewer.
+    this.set_('VIEWER', () => {
+      const viewerUrl = viewerFor(this.win_).getParam(VIEWPORT_ORIGIN_PARAM_) ||
```

Comment:
hmm. That complicates things if apps want to host a amp doc. How will apps do this handshake? 

For analytics, client id is also dependent on the origin being trusted. I'll start with just using the value of `messagingOrigin_`. Is that always set in case of google.com?